src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -61,30 +61,45 @@ private void captureIde() {
         if (!isCi()) {
             // Wait for projects to load to ensure Gradle project properties are initialized
             gradle.projectsEvaluated(g -> {
+                Optional<String> ideaVendorName = sysProperty("idea.vendor.name");
+                Optional<String> ideaVersion = sysProperty("idea.version");
                 Optional<String> invokedFromAndroidStudio = projectProperty("android.injected.invoked.from.ide");
                 Optional<String> androidStudioVersion = projectProperty("android.injected.studio.version");
-                Optional<String> ideaVersion = sysProperty("idea.version");
                 Optional<String> eclipseVersion = sysProperty("eclipse.buildId");
                 Optional<String> ideaSync = sysProperty("idea.sync.active");
                 if (ideaSync.isPresent()) {
                     buildScan.tag("IDE sync");
                 }
-                if (invokedFromAndroidStudio.isPresent()) {
-                    buildScan.tag("Android Studio");
-                    androidStudioVersion.ifPresent(v -> buildScan.value("Android Studio version", v));
+                if (ideaVendorName.isPresent()) {
+                    String ideaVendorNameValue = ideaVendorName.get();
+                    if(ideaVendorNameValue.equals("Google")) {
+                        // using androidStudioVersion instead of ideaVersion for compatibility reasons, those can be different (ie. 2020.3.1 Patch 3 instead of 2020.3)
+                        tagIde("Android Studio", androidStudioVersion.orElse(""));
+                    } else if(ideaVendorNameValue.equals("JetBrains")) {
+                        tagIde("IntelliJ IDEA", ideaVersion.orElse(""));
+                    }
+                } else if (invokedFromAndroidStudio.isPresent()) {
+                    // this case should be handled by the ideaVendorName condition but keeping it for compatibility reason (ideaVendorName started with 2020.1)
+                    tagIde("Android Studio", androidStudioVersion.orElse(""));
                 } else if (ideaVersion.isPresent()) {
-                    buildScan.tag("IntelliJ IDEA");
-                    buildScan.value("IntelliJ IDEA version", ideaVersion.get());
+                    // this case should be handled by the ideaVendorName condition but keeping it for compatibility reason (ideaVendorName started with 2020.1)
+                    tagIde("IntelliJ IDEA", ideaVersion.get());
                 } else if (eclipseVersion.isPresent()) {
-                    buildScan.tag("Eclipse");
-                    buildScan.value("Eclipse version", eclipseVersion.get());
+                    tagIde("Eclipse", eclipseVersion.get());
                 } else {
                     buildScan.tag("Cmd Line");
                 }
             });
         }
     }
 
+    private void tagIde(String ideLabel, String version) {
+        buildScan.tag(ideLabel);
+        if(!version.isEmpty()) {
```

Comment:
i think if we did the null check here we could remove the `orElse("")` and simplify all the callsites. WDYT?

Reply:
Do you mean passing an optional as method parameter (calling `get()` on a nulled Optional throws an exception)?
this is [discouraged](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html)

furthermore, doing that would test the optional parameter twice for Eclipse case and this [IntelliJ](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/91/files#diff-c2a2e523f8b7d89d2a35757da6898ae13ccc09fc7f735c03f820dc06f7956896R84) one

Reply:
I mean like this:
```suggestion
        if(version == null && !version.isEmpty()) {
```

Reply:
`version` wouldn't be null, a `NoSuchElementException` [would be raised](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#get())